### ISSUE TITLE
How do function comments relate to the function?

### ISSUE THREAD

**Issue by enygma** [psecio/parse#12]
In #10 there's a discussion about adding annotations to functions to tell Parse to bypass them to not include them in the results. It needs to be determined how the comments (with the annotation) relates to the function and where the best place would be to check this.

**Comment by redbeardcreator** [comment]
I just did some investigation, and things look really good. It seems `PhpParser` helps us significantly.

As long as you're using an actual DocBlock (e.g. `/** ... */`), each node that has a DocBlock will have it available via the `getDocComment()` method on the `Node`. The visitor can set flags when `enterNode()` is called if there is a DocBlock present that it recognizes (via parsing the DocBlock). To restore flags properly, the old version of the flags can be stored on the node itself via `setAttribute()`, then restored on `leaveNode()`. Basically:

```php
// ...
public function enterNode(Node $node)
{
    // ...
    if ($options = $this->parseDocBlock($node->getDocComment())) {
        $node->setAttribute('parseVisitorOptions', $this->options);
        $this->options = $options;
    }
    // ...
}

public function leaveNode($node)
{
    // ...
    if (null !== $options = $node->getAttribute('parseVisitorOptions')) {
        $this->options = $options;
    }
    // ...
}
```

The visitor options are set from the command line or configuration (or both) and individual DocBlocks can override the options as needed. This happens automatically at the class, method/function and statement level. I don't think it works at the file level, but that's not going to generally be a problem, I don't believe.

It's really a pretty simple system to manage. I've got some very experimental proof-of-concept code (that I won't share until I clean it up, since I'm doing way too many other things with it), that proves it works.

I'll work on it more once I get more of the ParseTest tests and revealed fixes done.
